Add composer v2 support

boesing opened the pull request:
|    Q          |   A
|-------------- | ------
| BC Break      | no
| New Feature   | yes

### Description

I've added composer 2.0 support. Not quite sure if we should go with `version_compare` in those parts or if we create own plugins depending on versions?

Any suggestions?

boesing committed:
Add composer v2 support

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Remove debug

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

michalbundyra commented on composer.json:
```diff
@@ -27,12 +27,13 @@
     "require-dev": {
-        "composer/composer": "^1.4.1",
+        "composer/composer": "^1.4.1 || ^2.0@dev",
+        "composer/semver": "^1.5.1 || ^2.0@dev",
```
@boesing we should now remove this, as we have already stable [semver 2.0](https://github.com/composer/semver/releases/tag/2.0.0).

boesing committed:
Remove `composer/semver` dependency

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Add CHANGELOG entry

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

samsonasik commented on composer.json:
```diff
@@ -27,12 +27,12 @@
     },
     "require-dev": {
-        "composer/composer": "^1.4.1",
+        "composer/composer": "^1.4.1 || ^2.0@dev",
```
probably use : 

```suggestion
        "composer/composer": "^1.4.1 || ~2.0.0@dev || ^2.0",
```

to make consistent with https://github.com/mezzio/mezzio-skeleton/blob/ed34e5870cb0ed49baffd7e458b8d056593b29bc/composer.json#L53

boesing committed:
Changed dev-dependency for `composer/composer`

Add composer 2.0.0 development compatibility and support for upcoming 2.x releases

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>
